Content.Shared/Clumsy/ClumsySystem.cs

```diff
@@ -0,0 +1,131 @@
+{
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly SharedStunSystem _stun = default!;
+    [Dependency] private readonly SharedAudioSystem _audioSystem = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+    [Dependency] private readonly DamageableSystem _damageableSystem = default!;
+    [Dependency] private readonly IGameTiming _timing = default!;
+    [Dependency] private readonly IConfigurationManager _cfg = default!;
+
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<ClumsyComponent, SelfBeforeHyposprayInjects>(BeforeHyposprayEvent);
+        SubscribeLocalEvent<ClumsyComponent, SelfBeforeDefibrillatorZaps>(BeforeDefibrillatorZapsEvent);
+        SubscribeLocalEvent<ClumsyComponent, SelfBeforeGunShotEvent>(BeforeGunShotEvent);
+        SubscribeLocalEvent<ClumsyComponent, SelfBeforeClimbEvent>(OnBeforeClimbEvent);
+    }
+
+    // If you add more clumsy interactions add them in this section!
+    #region Clumsy interaction events
+    private void BeforeHyposprayEvent(Entity<ClumsyComponent> entity, ref SelfBeforeHyposprayInjects args)
+    {
+        // Clumsy people sometimes inject themselves! Apparently syringes are clumsy proof...
+        if (!_random.Prob(entity.Comp.ClumsyDefaultCheck))
+            return;
+
+        args.TargetGettingInjected = args.EntityUsingHypospray;
+        args.InjectMessageOverride = "hypospray-component-inject-self-clumsy-message";
+        _audioSystem.PlayPvs(entity.Comp.ClumsySound, entity);
+    }
+
+    private void BeforeDefibrillatorZapsEvent(Entity<ClumsyComponent> entity, ref SelfBeforeDefibrillatorZaps args)
+    {
+        // Clumsy people sometimes defib themselves!
+        if (!_random.Prob(entity.Comp.ClumsyDefaultCheck))
+            return;
+
+        args.DefibTarget = args.EntityUsingDefib;
+        _audioSystem.PlayPvs(entity.Comp.ClumsySound, entity);
+
+    }
+
+    private void BeforeGunShotEvent(Entity<ClumsyComponent> entity, ref SelfBeforeGunShotEvent args)
+    {
+        // Clumsy people sometimes can't shoot :(
+
+        if (args.Gun.Comp.ClumsyProof == true)
+            return;
+
+        if (!_random.Prob(entity.Comp.ClumsyDefaultCheck))
+            return;
+
+        _stun.TryParalyze(entity, TimeSpan.FromSeconds(3f), true);
+
+        // Apply salt to the wound ("Honk!") (No idea what this comment means)
+        _audioSystem.PlayPvs(new SoundPathSpecifier("/Audio/Weapons/Guns/Gunshots/bang.ogg"), entity);
+        _audioSystem.PlayPvs(entity.Comp.ClumsySound, entity);
+
+        _popup.PopupEntity(Loc.GetString("gun-clumsy"), entity, entity);
+        args.Cancel();
+    }
+
+    private void OnBeforeClimbEvent(Entity<ClumsyComponent> entity, ref SelfBeforeClimbEvent args)
+    {
+        // This event is called in shared, thats why it has all the extra prediction stuff.
+        var rand = new System.Random((int)_timing.CurTick.Value);
```

Comment:
This is not a good way to do predicted randomness, since cheat clients could just delay the action to a tick when they get a favorable result.

Just return early on client instead of trying to predict this.

Reply:
Which maintainer said that this was fine?

Reply:
I don't think any cheat dev is even going to look / care about this, I'd rather just avoid spending time on it until we get a proper pseudorng system on engine.